File: 백준/Silver/5648. 역원소 정렬/역원소 정렬.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

    static int N;
    static long[] arr;
    static StringBuilder sb = new StringBuilder();

    static int stoi(String s) {
        return Integer.parseInt(s);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        N = stoi(st.nextToken());
        arr = new long[N];
        int i = 0;

        while (st.hasMoreTokens()) {
            arr[i++] = reverseParse(st.nextToken());
        }

        while (i < N) {
            st = new StringTokenizer(br.readLine());

            while (st.hasMoreTokens()) {
                arr[i++] = reverseParse(st.nextToken());
            }
        }

        Arrays.sort(arr);

        for (long x : arr) {
            sb.append(x).append("\n");
        }

        System.out.println(sb);
    }

    static long reverseParse(String str) {
        long num = 0;
        for (int i = str.length() - 1; i >= 0; i--) {
            num = num * 10 + (str.charAt(i) - '0');
        }
        return num;
    }
}

File: 백준/Silver/5648. 역원소 정렬/역원소 정렬.kt
import java.util.*

fun main() = with(System.`in`.bufferedReader()) {
    var st = StringTokenizer(readLine())
    val N = st.nextToken().toInt()

    val arr = LongArray(N)
    var idx = 0

    while (st.hasMoreTokens()) {
        arr[idx++] = reverseParse(st.nextToken())
    }

    while (idx < N) {
        st = StringTokenizer(readLine())
        
        while (st.hasMoreTokens()) {
            arr[idx++] = reverseParse(st.nextToken())
        }
    }

    arr.sort()

    print(buildString { arr.forEach { appendLine(it) } })
}

private fun reverseParse(str: String): Long {
    var num: Long = 0
    for (i in str.length - 1 downTo 0) {
        num = num * 10 + (str[i] - '0')
    }
    return num
}